File: presentation/src/main/java/com/khalturin/dmitriy/presentation/di/module/ApplicationModule.java
package com.khalturin.dmitriy.presentation.di.module;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 22.03.18 17:53.
 */

@Module
public class ApplicationModule {

  private Context mApplicationContext;

  public ApplicationModule(Context mApplicationContext){
    this.mApplicationContext = mApplicationContext;
  }

  @Provides
  @Singleton
  Context provideContext(){
    return this.mApplicationContext;
  }

}

File: rssreader/src/main/kotlin/com/khalturin/dmitriy/rssreader/di/module/ApplicationModule.kt
package com.khalturin.dmitriy.rssreader.di.module

import android.app.Application
import android.content.Context
import android.content.res.Resources
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 26.10.18 21:38.
 */
@Module
class ApplicationModule(private val mApplication: Application) {

  @Singleton
  @Provides
  fun provideContext(): Context = mApplication

  @Singleton
  @Provides
  fun provideResources(): Resources = mApplication.resources

}
